pkg/archive/changes.go

```diff
@@ -123,8 +127,16 @@ func Changes(layers []string, rw string) ([]Change, error) {
 							return nil
 						}
 					}
+					changedDirs[path] = struct{}{}
 					change.Kind = ChangeModify
 					break
+				} else if os.IsNotExist(err) && f.Mode().IsRegular() {
```

Comment:
Should this check be done outside of the loop over the layers? Meaning, should we only add this file's parent dir if we don't see it in ANY of the parent layers? Is it really safe to do it just based on the first layer we look at?